### PR TITLE
tdlib 1.3.0 (new formula)

### PR DESCRIPTION
**Pull request by bradtgmurray**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

https://core.telegram.org/tdlib

First contribution from me, please be patient. Thanks!

### CODE REVIEW

File: Formula/tdlib.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+class Tdlib < Formula
+  desc "Cross-platform library for building Telegram clients"
+  homepage "https://core.telegram.org/tdlib"
+  url "https://github.com/tdlib/td/archive/v1.3.0.tar.gz"
+  sha256 "2953fe75027ac531248b359123aa4812666377ac874c1db506fa74176f2d2338"
+  head "https://github.com/tdlib/td.git"
```
Comment (fxcoudert) [inline Formula/tdlib.rb]: No `head`

File: Formula/tdlib.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "gperf"
+  depends_on "openssl"
+  depends_on "readline"
+
+  def install
+    mkdir "build" do
+      system "cmake", "-DCMAKE_BUILD_TYPE=Release", "-DOPENSSL_ROOT_DIR=/usr/local/Cellar/openssl/1.0.2p", ".."
```
Comment (fxcoudert) [inline Formula/tdlib.rb]: - `CMAKE_BUILD_TYPE` is part of `std_cmake_args`
- Why are your running `cmake` twice?
- Can cmake find openssl without the special flag?

File: Formula/tdlib.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      project(TdExample VERSION 1.0 LANGUAGES CXX)
+
+      find_package(Td 1.3.0 REQUIRED)
+
+      add_executable(tdjson_example tdjson_example.cpp)
+      target_link_libraries(tdjson_example PRIVATE Td::TdJson)
+      set_property(TARGET tdjson_example PROPERTY CXX_STANDARD 11)
+    EOS
+
+    system "cmake", "."
```
Comment (fxcoudert) [inline Formula/tdlib.rb]: We prefer tests that don't need `cmake`, calling `system ENV.cxx, "test.cpp", "-L…", "-l…", etc`

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/tdlib.rb (hunk 2 above)
Comment (bradtgmurray) [reply Formula/tdlib.rb]: Re: cmake twice

Following the instructions in the parent project: https://github.com/tdlib/td/blob/master/README.md#building

Note the change from .. to ., it's almost like the project is set up to generate a cmake project in the build directory and then from there generate a makefile.

I was seeing link errors without the OPENSSL_ROOT_DIR define, see the commentary at https://github.com/tdlib/td/blob/master/README.md#macos

However, this feels really gross to be hardcoding the version of openssl and the path to my Cellar directory, are there some variables I should be using instead?

File: Formula/tdlib.rb (hunk 1 above)
Comment (bradtgmurray) [reply Formula/tdlib.rb]: 👍

File: Formula/tdlib.rb (hunk 3 above)
Comment (bradtgmurray) [reply Formula/tdlib.rb]: 👍

File: Formula/tdlib.rb (hunk 2 above)
Comment (fxcoudert) [reply Formula/tdlib.rb]: `#{Formula["openssl"].opt_prefix}`

Comment (bradtgmurray) [reply Formula/tdlib.rb]: :+1:

Comment (bradtgmurray) [reply Formula/tdlib.rb]: Actually, looks like this wasn't necessary after all, my bad.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @bradtgmurray for the pull request!

**Pull request closed without merging**